src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,320 @@
+                                                         struct PAL_GssBuffer* outBuffer)
+{
+    return NetSecurityNative_DisplayStatus(minorStatus, statusValue, GSS_C_GSS_CODE, outBuffer);
+}
+
+extern "C" uint32_t
+NetSecurityNative_ImportUserName(uint32_t* minorStatus, char* inputName, uint32_t inputNameLen, GssName** outputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+    assert(outputName != nullptr);
+
+    GssBuffer inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_C_NT_USER_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurityNative_ImportPrincipalName(uint32_t* minorStatus,
+                                                          char* inputName,
+                                                          uint32_t inputNameLen,
+                                                          GssName** outputName)
+{
+    assert(minorStatus != nullptr);
+    assert(inputName != nullptr);
+    assert(outputName != nullptr);
+
+    GssBuffer inputNameBuffer{.length = inputNameLen, .value = inputName};
+    gss_OID nameType = const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t NetSecurityNative_InitSecContext(uint32_t* minorStatus,
+                                                     GssCredId* claimantCredHandle,
+                                                     GssCtxId** contextHandle,
+                                                     uint32_t isNtlm,
+                                                     GssName* targetName,
+                                                     uint32_t reqFlags,
+                                                     uint8_t* inputBytes,
+                                                     uint32_t inputLength,
+                                                     struct PAL_GssBuffer* outBuffer,
+                                                     uint32_t* retFlags)
+{
+    assert(minorStatus != nullptr);
+    assert(contextHandle != nullptr);
+    assert(isNtlm == 0 || isNtlm == 1);
+    assert(targetName != nullptr);
+    assert(inputBytes != nullptr || inputLength == 0);
+    assert(outBuffer != nullptr);
+    assert(retFlags != nullptr);
+
+// Note: claimantCredHandle can be null
+
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID desiredMech = isNtlm ? GSS_NTLM_MECHANISM : GSS_SPNEGO_MECHANISM;
+#else
+    assert(!isNtlm && "NTLM is not supported by MIT libgssapi_krb5");
+    (void)isNtlm; // unused
+
+    gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_spnego_oid_value)};
+    gss_OID desiredMech = &gss_mech_spnego_OID_desc;
+#endif
+
+    GssBuffer inputToken{.length = UnsignedCast(inputLength), .value = inputBytes};
```

Comment:
In `NetSecurityNative_ImportPrincipalName` the `inputNameLen` value (`uint32_t`) was directly assigned.  Here you're using `UnsignedCast` for another `uint32_t`.  Something seems a bit off...